### DISCUSSION TITLE
Typst Default Template: Parameter Level for "section-numbering"

### DISCUSSION THREAD

**Discussion by 7enderhead** [jgm/pandoc#10257] [General]
The current Typst default template reads the Typst-specific section numbering scheme from a top-level parameter `section-numbering`:

YAML:
```yaml
author: Joe Doe
section-numbering: "1.a"
```

Template starting from line 84:

```
$if(section-numbering)$
  sectionnumbering: "$section-numbering$",
$endif$
```

What is the logic behind that? I think of this parameter as having a Typst-specific format, e.g. "1.a" to have numbers on first-level headings and letters on second-level headings.

Wouldn't a Typst-specific parameter make more sense?

```yaml
---
author: Joe Doe
output:
  typst:
    section-numbering: "1.a"
```

Template:

```
$if(output.typst.section-numbering)$
  sectionnumbering: "$output.typst.section-numbering$",
$endif$
```

**Comment by jgm** [comment]
It's just a template variable.  Not every template variable is used in every template.  You're putting it metadata, but personally I like to separate metadata from things like this (especially when they are output-specific); if you want to separate it you can specify it on the command line `-Vsection-numbering='i.a` or in a defaults file.
